alacritty_terminal/src/term/color.rs

```diff
@@ -127,7 +134,12 @@ impl FromStr for Rgb {
 
         component!(r, g, b);
 
-        Ok(rgb)
+        if chars.as_str().is_empty() {
+            Ok(rgb)
+        } else {
+            // We still have something to parse, so throw an error
+            Err(())
+        }
```

Comment:
I wonder if it wouldn't be much easier to check the length before starting the parsing.

That way we can just skip a lot of things.

I was thinking something like this, wrt method structure:

```rust
let chars = if input.starts_with("0x") && input.len() == 8 {
} else if input.starts_with("#") && input.len() == 7 {
} else {
     return Err(());
};
```

Then we can potentially even clean up the macro a bit, or remove it completely. We can also change it from a chars iterator to a slice reference, by returning something like `&input[1..]`, because that way we can directly use `slice[1]` to index it, instead of having to use `chars.next()`.

Just some ideas though, I think it would be possible to simplify and shorten things up a bit.